### PR TITLE
Fix combined posts which ignores custom and other post types

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Fix combined posts which ignores custom and other post types

#### Ticket Link
None but posted here - https://pre-release.mattermost.com/core/pl/e7kgqmhjhtr5iyt3qyik5e6qcw

#### Checklist
- [x] Added or updated unit tests (required for all new features)

#### Test Information
This PR was tested on: [MacOS/Chrome/FF, iOS simulator]

### CODE REVIEW

File: test/utils/post_utils.test.js [hunk 1]
```diff
@@ -1079,5 +1085,33 @@ describe('PostUtils', () => {
...
+                allUserIds: ['user_id_2', 'added_user_id_1', 'user_id_1'],
+                messageData: [
+                    {postType: 'system_join_channel', userIds: ['user_id_2']},
+                    {postType: 'system_add_to_channel', userIds: ['added_user_id_1'], actorId: 'user_id_1'},
+                ],
+            };
+
+            const combinedPostId = out.postsForChannel[1];
+            console.log('out.nextPosts[combinedPostId]:', out.nextPosts[combinedPostId]);
+            console.log('out.postsForChannel:', out.postsForChannel);
```
Comment (jwilander) [inline test/utils/post_utils.test.js]: Are these logs needed or just for your debugging?

File: test/utils/post_utils.test.js [hunk 2]
```diff
@@ -1079,5 +1085,33 @@ describe('PostUtils', () => {
...
+            assert.equal(out.postsForChannel.length, 4);
```
Comment (jwilander) [inline test/utils/post_utils.test.js]: Can you add an explicit assert to make sure that the custom post is included properly?

File: test/utils/post_utils.test.js (hunk 1 above)
Comment (saturninoabril) [reply test/utils/post_utils.test.js]: Ooppss, now removed.

File: test/utils/post_utils.test.js (hunk 2 above)
Comment (saturninoabril) [reply test/utils/post_utils.test.js]: Added

**Pull request merged**